web-server/pages/api/resources/orgs/[org_id]/teams/v2.ts

```diff
@@ -202,22 +202,26 @@ endpoint.handle.DELETE(deleteSchema, async (req, res) => {
     .returning('*')
     .then(getFirstRow);
 
-  const activeTeamReposIds = await db('TeamRepos')
+  const inactiveTeamReposIds: string[] = await db('TeamRepos')
     .update('is_active', false)
     .where('team_id', req.payload.id)
     .andWhere('is_active', true)
     .returning('org_repo_id')
     .then((result) => result.map((row) => row.org_repo_id));
```

Comment:
@Kamlesh72 `andWhere('is_active', true)` will return active team repos, so why you have named this array as `inactiveTeamReposIds`

Reply:
@shivam-bit All the active teamRepos will be marked inactive eventually. 

Hence inactiveRepoIds - activeRepoIds = repos that needs to be inactivate.

Reply:
@Kamlesh72 
> Hence, inactiveRepoIds - activeRepoIds = repos that need to be inactivated.

1. Don't you think it should be `activeRepoIds - inactiveRepoIds = repos` that are active but need to be marked as inactive?

> All the active teamRepos will eventually be marked inactive.
2. In that case, `inactiveTeamRepoIds` is confusing, as it represents something that hasn’t been done yet.


PS: It might be helpful to add comments, as this part can be a bit confusing.

Reply:
@shivam-bit 

1. inactiveTeamReposIds represent repos of team which we updated to inactive (all repos of this team will be inactive since team is deleted).

Now we want to check if there is any repo which is still being used in other teams. If we get repos which are still used in other team, that needs to be removed from inactiveTeamReposIds.

So (inactive/all repos of this team) - (active repos from other teams) = repos which are nowhere used.

```javascript
const orgReposToBeInactivate = inactiveTeamReposIds.filter(
  (id) => !activeOrgReposIds.includes(id)
);
```

2. I will add the required comments.

Reply:
@shivam-bit  Is this fine?

```javascript
// 1. Mark inactive and Get Repo Ids of this deleted team. Ex: [ '123', '456', '789' ]
const deletedTeamRepoIds: string[] = await db('TeamRepos')
    .update('is_active', false)
    .where('team_id', req.payload.id)
    .andWhere('is_active', true)
    .returning('org_repo_id')
    .then((result) => result.map((row) => row.org_repo_id));

// 2. Get Repo Ids which are still used across other teams. Ex: [ '456', '789' ]
const activeRepoIds: string[] = await db('TeamRepos')
    .where('is_active', true)
    .whereIn('org_repo_id', inactiveTeamReposIds)
    .distinct('org_repo_id')
    .then((result) => result.map((row) => row.org_repo_id));

// 3. Repo Ids which are nowhere used. Ex: [ '123' ]
const orgRepoIdsToBeInactivate = deletedTeamRepoIds.filter(
    (id) => !activeRepoIds.includes(id)
);

await db('OrgRepo')
    .update('is_active', false)
    .whereIn('id', orgRepoIdsToBeInactivate);
```